pkg/splunk/enterprise/util.go

```diff
@@ -882,3 +866,85 @@ func setupAppInitContainers(client splcommon.ControllerClient, cr splcommon.Meta
 		}
 	}
 }
+
+// SetLastAppInfoCheckTime sets the last check time to current time
+func SetLastAppInfoCheckTime(appInfoStatus *enterprisev1.AppDeploymentContext) {
+	scopedLog := log.WithName("SetLastAppInfoCheckTime")
+	currentEpoch := time.Now().Unix()
+
+	scopedLog.Info("Setting the LastAppInfoCheckTime to current time", "current epoch time", currentEpoch)
+
+	appInfoStatus.LastAppInfoCheckTime = currentEpoch
+}
+
+// HasAppRepoCheckTimerExpired checks if the polling interval has expired
+func HasAppRepoCheckTimerExpired(appFrameworkRef *enterprisev1.AppFrameworkSpec, appInfoContext *enterprisev1.AppDeploymentContext) bool {
+	scopedLog := log.WithName("hasAppRepoCheckTimerExpired")
+	currentEpoch := time.Now().Unix()
+	scopedLog.Info("Checking if the app repo polling interval has expired", "LastAppInfoCheckTime", strconv.FormatInt(appInfoContext.LastAppInfoCheckTime, 10), "current epoch time", strconv.FormatInt(currentEpoch, 10))
+	return appInfoContext.LastAppInfoCheckTime+int64(appFrameworkRef.AppsRepoPollInterval) <= currentEpoch
+}
```

Comment:
Nit: To avoid excessive logging across the reconciles, we can just log when it is true